monoio-http/src/h1/codec/decoder.rs

```diff
@@ -177,6 +190,43 @@ impl Decoder for RequestHeadDecoder {
     }
 }
 
+fn normalize_path(path: &str) -> String {
+    let mut result = String::with_capacity(path.len());
+
+    // Add leading slash if needed
+    if !path.starts_with('/') {
+        result.push('/');
+    }
+
+    // Check if path ends with slash
+    let has_trailing_slash = path.ends_with('/');
+
+    // Split path into segments and process them
+    let segments: Vec<&str> = path.split('/').filter(|s| !s.is_empty()).collect();
+    let mut normalized_segments = Vec::with_capacity(segments.len());
+
+    for segment in segments {
+        match segment {
+            "." => continue,
+            ".." => {
```

Comment:
Treating `..` as going to upper directory and _normalizing_ it, is it always the thing that users want, or safe?